classes/class.ilMumieTaskGradeSync.php

```diff
@@ -136,24 +141,37 @@ private function getLastSync()
         if ($oldest_timestamp == PHP_INT_MAX) {
             $oldest_timestamp = 1;
         }
-        return $oldest_timestamp * 1000;
+        return $oldest_timestamp*1000;
     }
 
     /**
      * Get all users that can get marks for this MUMIE task
      */
     private function getAllUsers($task)
     {
-        global $ilDB;
-        $users = array();
-        $result = $ilDB->query("SELECT usr_id" .
-            " FROM ut_lp_marks" .
-            " WHERE obj_id = " . $ilDB->quote($task->getId(), "integer"));
+        if ($this->isInBaseRepository($task)) {
+            return ilParticipants::getInstance($task->getParentRef())->getMembers();
+        } else {
+            return $this->getAllUserIds();
+        }
+    }
 
-        while ($record = $ilDB->fetchAssoc($result)) {
-            array_push($users, $record['usr_id']);
+    private function isInBaseRepository($task)
+    {
+        return $task->getParentRef() != 1;
+    }
```

Comment:
I think this method is misnamed. From what I can tell, it returns truem if something is NOT in the BaseRepository

Reply:
not